File: frontend/src/lib/audit.ts
export interface ScoreBreakdown {
  id: string;
  score_key: string;
  label: string;
  name: string;
  score: number;
  weight: number;
  description: string;
  details: string;
  findings_count: number;
}

export interface AuditFinding {
  id: string;
  rule_id: string;
  severity: "critical" | "high" | "medium" | "low" | string;
  category: string;
  title: string;
  description: string;
  entity_type: string;
  entity_id: string;
  entity_name: string;
  affected_entity: string;
  affected_entity_id: string;
  metric_value: number | null;
  threshold_value: number | null;
  estimated_waste: number;
  estimated_uplift: number;
  recommendation_key: string | null;
  score_impact: number;
}

export interface Recommendation {
  id: string;
  recommendation_key: string;
  title: string;
  body: string;
}

export interface AuditAISummary {
  id: string;
  audit_run_id: string;
  provider: string;
  model: string;
  prompt_version: string;
  status: "completed" | "failed" | "pending" | string;
  short_executive_summary: string;
  detailed_audit_explanation: string;
  prioritized_action_plan: string;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface AuditJob {
  job_id: string;
  status: string;
}

export interface AuditJobStatus {
  job_id: string;
  status: string;
  error?: string | null;
  completed_audit_id?: string | null;
}

export interface AuditReport {
  id: string;
  health_score: number;
  total_spend: number;
  total_wasted_spend: number;
  total_estimated_uplift: number;
  findings_count: number;
  analysis_start: string;
  analysis_end: string;
  created_at: string;
  campaign_count: number;
  ad_set_count: number;
  ad_count: number;
  findings: AuditFinding[];
  scores: ScoreBreakdown[];
  pillar_scores: ScoreBreakdown[];
  recommendations: Recommendation[];
  ai_summary: AuditAISummary | null;
  job_status: string;
  job_error: string | null;
  celery_task_id: string | null;
}

export interface SeverityCounts {
  critical: number;
  high: number;
  medium: number;
  low: number;
}

export interface TrendPoint {
  label: string;
  spend: number;
  roas: number;
}

export interface LeaderboardItem {
  entity_id: string;
  entity_name: string;
  spend: number;
  roas: number;
  cpa: number;
  ctr: number;
}

export interface MetricSplit {
  key: string;
  label: string;
  value: number;
}

export interface AccountKpis {
  spend: number;
  impressions: number;
  reach: number;
  clicks: number;
  ctr: number;
  cpc: number;
  cpm: number;
  conversions: number;
  conversion_value: number;
  roas: number;
  frequency: number;
  cpa: number;
  wow_spend_delta: number;
  wow_ctr_delta: number;
  wow_roas_delta: number;
  wow_cpa_delta: number;
  daily_budget_total: number;
  lifetime_budget_total: number;
  objective_mix: MetricSplit[];
  optimization_goal_mix: MetricSplit[];
  status_mix: MetricSplit[];
}

export interface AuditDashboardData {
  audit: AuditReport | null;
  kpis: AccountKpis;
  data_mode: string;
  limitations: string[];
  severity_counts: SeverityCounts;
  top_opportunities: AuditFinding[];
  spend_roas_trend: TrendPoint[];
  worst_campaigns: LeaderboardItem[];
  worst_adsets: LeaderboardItem[];
}

export interface AuditSummary {
  id: string;
  health_score: number;
  total_spend: number;
  total_wasted_spend: number;
  total_estimated_uplift: number;
  findings_count: number;
  analysis_start: string;
  analysis_end: string;
  created_at: string;
}

export function formatCurrency(value: number): string {
  return "$" + value.toLocaleString(undefined, { minimumFractionDigits: 0, maximumFractionDigits: 0 });
}

export function formatPercent(value: number): string {
  return `${value.toFixed(2)}%`;
}

export function formatDate(value: string): string {
  return new Date(value).toLocaleDateString(undefined, {
    month: "short",
    day: "numeric",
    year: "numeric",
  });
}

export function cleanAiSummaryText(text: string | null | undefined): string | null {
  if (!text) return null;

  return (
    text
      .replace(/AI generation fallback used \([^)]+\)\.?/gi, "")
      .replace(/provider not configured\.?/gi, "")
      .replace(/\s{2,}/g, " ")
      .trim() || null
  );
}

File: backend/app/schemas/audit.py
from datetime import date, datetime

from pydantic import BaseModel


class AuditFindingResponse(BaseModel):
    id: str
    rule_id: str
    severity: str
    category: str
    title: str
    description: str
    entity_type: str
    entity_id: str
    entity_name: str
    affected_entity: str
    affected_entity_id: str
    metric_value: float | None
    threshold_value: float | None
    estimated_waste: float
    estimated_uplift: float
    recommendation_key: str | None
    score_impact: float
    confidence_label: str
    confidence_reason: str
    inspection_target: str


class RecommendationResponse(BaseModel):
    id: str
    audit_finding_id: str | None = None
    recommendation_key: str
    title: str
    body: str

    model_config = {"from_attributes": True}


class AuditScoreResponse(BaseModel):
    id: str
    score_key: str
    label: str
    name: str
    score: float
    weight: float
    description: str
    details: str
    findings_count: int = 0
    strongest_issue: str | None = None


class TrendPointResponse(BaseModel):
    label: str
    spend: float
    roas: float


class LeaderboardItemResponse(BaseModel):
    entity_id: str
    entity_name: str
    spend: float
    roas: float
    cpa: float
    ctr: float


class SeverityCountResponse(BaseModel):
    critical: int
    high: int
    medium: int
    low: int


class MetricSplitResponse(BaseModel):
    key: str
    label: str
    value: float


class AccountKpiResponse(BaseModel):
    spend: float
    impressions: int
    reach: int
    clicks: int
    ctr: float
    cpc: float
    cpm: float
    conversions: int
    conversion_value: float
    roas: float
    frequency: float
    cpa: float
    wow_spend_delta: float
    wow_ctr_delta: float
    wow_roas_delta: float
    wow_cpa_delta: float
    daily_budget_total: float
    lifetime_budget_total: float
    objective_mix: list[MetricSplitResponse]
    optimization_goal_mix: list[MetricSplitResponse]
    status_mix: list[MetricSplitResponse]


class AuditAISummaryResponse(BaseModel):
    id: str
    audit_run_id: str
    provider: str
    model: str
    prompt_version: str
    status: str
    short_executive_summary: str
    detailed_audit_explanation: str
    prioritized_action_plan: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class AuditRunResponse(BaseModel):
    id: str
    health_score: float
    total_spend: float
    total_wasted_spend: float
    total_estimated_uplift: float
    findings_count: int
    analysis_start: date
    analysis_end: date
    created_at: datetime
    campaign_count: int
    ad_set_count: int
    ad_count: int
    findings: list[AuditFindingResponse]
    scores: list[AuditScoreResponse]
    pillar_scores: list[AuditScoreResponse]
    recommendations: list[RecommendationResponse]
    ai_summary: AuditAISummaryResponse | None = None
    job_status: str = "completed"
    job_error: str | None = None
    celery_task_id: str | None = None


class AuditDashboardResponse(BaseModel):
    audit: AuditRunResponse | None
    kpis: AccountKpiResponse
    data_mode: str = "daily_breakdown"
    limitations: list[str] = []
    severity_counts: SeverityCountResponse
    top_opportunities: list[AuditFindingResponse]
    spend_roas_trend: list[TrendPointResponse]
    worst_campaigns: list[LeaderboardItemResponse]
    worst_adsets: list[LeaderboardItemResponse]


class AuditSummaryResponse(BaseModel):
    id: str
    health_score: float
    total_spend: float
    total_wasted_spend: float
    total_estimated_uplift: float
    findings_count: int
    analysis_start: date
    analysis_end: date
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditJobResponse(BaseModel):
    job_id: str
    status: str


class AuditJobStatusResponse(BaseModel):
    job_id: str
    status: str
    error: str | None = None
    completed_audit_id: str | None = None
